From 4bae5006c9bf5af5fcbd19264ac6c015d2506240 Mon Sep 17 00:00:00 2001
From: Mario Sanchez Prada <mario@endlessm.com>
Date: Wed, 4 Jul 2018 17:02:26 +0100
Subject: [PATCH 1/4] ffmpeg: Use royalty-free libfdk-aac for AAC playback

Changes:
- Explicitly add `ff_fdk_aac_decoder` to the list of codecs provided
  by the internal libffmpeg build when using the Chromium (free codecs)
  configuration, so that Chromium will use the libfdk-aac implementation
  for AAC playback on free builds.
- Also replace calls to `ff_aac_decoder` with `ff_fdk_aac_decoder` when using
  the Chrome (non-free codecs) configuration, so that Chromium also uses
  libfdk-aac for AAC playback on non-free builds (with extra codecs pack).
- Also enable the AAC demuxer in the Chromium (free) configuration,
  as is already the case for the Chrome (non-free) configuration, to fix
  an issue where chromium would fail to recognize/parse some AAC audio
  content without the demuxer enabled.
- Adapt `build_ffmpeg.py` to make sure it always enables the AAC decoder
  based on libfdk-aac and use it according to the instructions in [1]
  to update the different configurations we might be interest in.
- Note that we're also removing several files from the list of sources files,
  to prevent references to the non-free AAC decoder from being considered while
  linking, as that would likely cause the inclusion of the `ff_aac_decoder`
  symbol in the libffmpeg shared library, which should only contain
  references to the symbol `ff_fdk_aac_decoder`.

[1] https://github.com/endlessm/chromium-browser/commit/a7056234

Squash the following commits from v83 with slight changes to drop ARM
specific changes:
- ffmpeg: Specify sources and link flags required to build libfdk-aac binaries
- ffmpeg: Adapt script to generate ffmpeg configuration and update files
- ffmpeg: Fix AAC playback on free version
- ffmpeg: Remove references to ff_aac_decoder and add ff_fdk_aac_decoder instead

https://phabricator.endlessm.com/T21886
https://phabricator.endlessm.com/T30286
https://phabricator.endlessm.com/T30300
---
 third_party/ffmpeg/BUILD.gn                   |  17 +
 .../config/Chrome/linux/arm64/config.h        |   6 +-
 .../linux/arm64/libavcodec/codec_list.c       |   2 +-
 .../config/Chrome/linux/x64/config.asm        |   6 +-
 .../chromium/config/Chrome/linux/x64/config.h |   6 +-
 .../Chrome/linux/x64/libavcodec/codec_list.c  |   2 +-
 .../config/Chromium/linux/arm64/config.h      |  24 +-
 .../linux/arm64/libavcodec/bsf_list.c         |   1 +
 .../linux/arm64/libavcodec/codec_list.c       |   2 +
 .../linux/arm64/libavcodec/parser_list.c      |   1 +
 .../linux/arm64/libavformat/demuxer_list.c    |   1 +
 .../linux/arm64/libavutil/ffversion.h         |   2 +-
 .../config/Chromium/linux/x64/config.asm      |  26 +-
 .../config/Chromium/linux/x64/config.h        |  30 +-
 .../Chromium/linux/x64/libavcodec/bsf_list.c  |   1 +
 .../linux/x64/libavcodec/codec_list.c         |   2 +
 .../linux/x64/libavcodec/parser_list.c        |   1 +
 .../linux/x64/libavformat/demuxer_list.c      |   1 +
 .../Chromium/linux/x64/libavutil/ffversion.h  |   2 +-
 .../ffmpeg/chromium/scripts/build_ffmpeg.py   |  17 +-
 third_party/ffmpeg/ffmpeg_generated.gni       | 423 ++++--------------
 21 files changed, 172 insertions(+), 401 deletions(-)

diff --git a/third_party/ffmpeg/BUILD.gn b/third_party/ffmpeg/BUILD.gn
index f7f34ea414..337e45cb3f 100755
--- a/third_party/ffmpeg/BUILD.gn
+++ b/third_party/ffmpeg/BUILD.gn
@@ -435,6 +435,23 @@ if (is_component_ffmpeg) {
         "-L",
         rebase_path(target_gen_dir, root_build_dir),
       ]
+
+      # Need to link against the system's libfdk-aac.
+      if (current_cpu == "arm" || current_cpu == "arm64") {
+        ldflags += [
+          "-L",
+          "/usr/lib/arm-linux-gnueabihf/",
+        ]
+      } else {
+        ldflags += [
+          "-L",
+          "/usr/lib/x86_64-linux-gnu/",
+        ]
+      }
+      libs = [
+        "fdk-aac","openh264"
+      ]
+
     } else if (is_win) {
       # Windows component build needs the .def file to export ffmpeg symbols.
       deps += [ ":ffmpeg_generate_def" ]
diff --git a/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/config.h b/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/config.h
index 17cf7da560..b05203475c 100644
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/config.h
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/config.h
@@ -425,7 +425,7 @@
 #define CONFIG_LIBXAVS2 0
 #define CONFIG_LIBXVID 0
 #define CONFIG_DECKLINK 0
-#define CONFIG_LIBFDK_AAC 0
+#define CONFIG_LIBFDK_AAC 1
 #define CONFIG_OPENSSL 0
 #define CONFIG_LIBTLS 0
 #define CONFIG_GMP 0
@@ -1013,7 +1013,7 @@
 #define CONFIG_ZEROCODEC_DECODER 0
 #define CONFIG_ZLIB_DECODER 0
 #define CONFIG_ZMBV_DECODER 0
-#define CONFIG_AAC_DECODER 1
+#define CONFIG_AAC_DECODER 0
 #define CONFIG_AAC_FIXED_DECODER 0
 #define CONFIG_AAC_LATM_DECODER 0
 #define CONFIG_AC3_DECODER 0
@@ -1241,7 +1241,7 @@
 #define CONFIG_LIBCODEC2_DECODER 0
 #define CONFIG_LIBDAV1D_DECODER 0
 #define CONFIG_LIBDAVS2_DECODER 0
-#define CONFIG_LIBFDK_AAC_DECODER 0
+#define CONFIG_LIBFDK_AAC_DECODER 1
 #define CONFIG_LIBGSM_DECODER 0
 #define CONFIG_LIBGSM_MS_DECODER 0
 #define CONFIG_LIBILBC_DECODER 0
diff --git a/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/libavcodec/codec_list.c b/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/libavcodec/codec_list.c
index 49f757b2d8..7e49e07bbe 100644
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/libavcodec/codec_list.c
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/arm64/libavcodec/codec_list.c
@@ -3,7 +3,7 @@ static const AVCodec * const codec_list[] = {
     &ff_theora_decoder,
     &ff_vp3_decoder,
     &ff_vp8_decoder,
-    &ff_aac_decoder,
+    &ff_libfdk_aac_decoder,
     &ff_flac_decoder,
     &ff_mp3_decoder,
     &ff_vorbis_decoder,
diff --git a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.asm b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.asm
index 567fae94aa..c49be96217 100644
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.asm
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.asm
@@ -410,7 +410,7 @@
 %define CONFIG_LIBXAVS2 0
 %define CONFIG_LIBXVID 0
 %define CONFIG_DECKLINK 0
-%define CONFIG_LIBFDK_AAC 0
+%define CONFIG_LIBFDK_AAC 1
 %define CONFIG_OPENSSL 0
 %define CONFIG_LIBTLS 0
 %define CONFIG_GMP 0
@@ -998,7 +998,7 @@
 %define CONFIG_ZEROCODEC_DECODER 0
 %define CONFIG_ZLIB_DECODER 0
 %define CONFIG_ZMBV_DECODER 0
-%define CONFIG_AAC_DECODER 1
+%define CONFIG_AAC_DECODER 0
 %define CONFIG_AAC_FIXED_DECODER 0
 %define CONFIG_AAC_LATM_DECODER 0
 %define CONFIG_AC3_DECODER 0
@@ -1226,7 +1226,7 @@
 %define CONFIG_LIBCODEC2_DECODER 0
 %define CONFIG_LIBDAV1D_DECODER 0
 %define CONFIG_LIBDAVS2_DECODER 0
-%define CONFIG_LIBFDK_AAC_DECODER 0
+%define CONFIG_LIBFDK_AAC_DECODER 1
 %define CONFIG_LIBGSM_DECODER 0
 %define CONFIG_LIBGSM_MS_DECODER 0
 %define CONFIG_LIBILBC_DECODER 0
diff --git a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.h b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.h
index 40a746aa5e..b9ed931b11 100644
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.h
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/config.h
@@ -425,7 +425,7 @@
 #define CONFIG_LIBXAVS2 0
 #define CONFIG_LIBXVID 0
 #define CONFIG_DECKLINK 0
-#define CONFIG_LIBFDK_AAC 0
+#define CONFIG_LIBFDK_AAC 1
 #define CONFIG_OPENSSL 0
 #define CONFIG_LIBTLS 0
 #define CONFIG_GMP 0
@@ -1013,7 +1013,7 @@
 #define CONFIG_ZEROCODEC_DECODER 0
 #define CONFIG_ZLIB_DECODER 0
 #define CONFIG_ZMBV_DECODER 0
-#define CONFIG_AAC_DECODER 1
+#define CONFIG_AAC_DECODER 0
 #define CONFIG_AAC_FIXED_DECODER 0
 #define CONFIG_AAC_LATM_DECODER 0
 #define CONFIG_AC3_DECODER 0
@@ -1241,7 +1241,7 @@
 #define CONFIG_LIBCODEC2_DECODER 0
 #define CONFIG_LIBDAV1D_DECODER 0
 #define CONFIG_LIBDAVS2_DECODER 0
-#define CONFIG_LIBFDK_AAC_DECODER 0
+#define CONFIG_LIBFDK_AAC_DECODER 1
 #define CONFIG_LIBGSM_DECODER 0
 #define CONFIG_LIBGSM_MS_DECODER 0
 #define CONFIG_LIBILBC_DECODER 0
diff --git a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c
index 49f757b2d8..7e49e07bbe 100644
--- a/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c
+++ b/third_party/ffmpeg/chromium/config/Chrome/linux/x64/libavcodec/codec_list.c
@@ -3,7 +3,7 @@ static const AVCodec * const codec_list[] = {
     &ff_theora_decoder,
     &ff_vp3_decoder,
     &ff_vp8_decoder,
-    &ff_aac_decoder,
+    &ff_libfdk_aac_decoder,
     &ff_flac_decoder,
     &ff_mp3_decoder,
     &ff_vorbis_decoder,
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/config.h b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/config.h
index 3d152af12a..4b10b2dc08 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/config.h
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/config.h
@@ -1,12 +1,12 @@
 /* Automatically generated by configure - do not modify! */
 #ifndef FFMPEG_CONFIG_H
 #define FFMPEG_CONFIG_H
-/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-fft --enable-rdft --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-lzo --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/mnt/src/chromium/src/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --enable-decoder='theora,vp8' --enable-parser='vp3,vp8' --enable-cross-compile --cross-prefix=/usr/bin/aarch64-linux-gnu- --target-os=linux --extra-cflags='--target=aarch64-linux-gnu' --extra-ldflags='--target=aarch64-linux-gnu' --sysroot=/mnt/src/chromium/src/build/linux/debian_sid_arm64-sysroot --arch=aarch64 --enable-armv8 --extra-cflags='-march=armv8-a' --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld'" -- elide long configuration string from binary */
+/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-fft --enable-rdft --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-lzo --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/chromium/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --enable-decoder='theora,vp8' --enable-parser='vp3,vp8' --enable-cross-compile --cross-prefix=/usr/bin/aarch64-linux-gnu- --target-os=linux --extra-cflags='--target=aarch64-linux-gnu' --extra-ldflags='--target=aarch64-linux-gnu' --sysroot=/chromium/build/linux/debian_sid_arm64-sysroot --arch=aarch64 --enable-armv8 --extra-cflags='-march=armv8-a' --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld' --enable-libfdk-aac --enable-decoder=libfdk_aac --enable-demuxer=aac --enable-parser=aac --enable-libopenh264 --enable-decoder=libopenh264" -- elide long configuration string from binary */
 #define FFMPEG_LICENSE "LGPL version 2.1 or later"
 #define CONFIG_THIS_YEAR 2020
 #define FFMPEG_DATADIR "/usr/local/share/ffmpeg"
 #define AVCONV_DATADIR "/usr/local/share/ffmpeg"
-#define CC_IDENT "clang version 12.0.0 (https://github.com/llvm/llvm-project/ d0abc757495349fd053beeaea81cd954c2e457e7)"
+#define CC_IDENT "clang version 12.0.0 (https://github.com/llvm/llvm-project/ 418f18c6cdfe56e77669e2f4d3df3bca1020156d)"
 #define av_restrict restrict
 #define EXTERN_PREFIX ""
 #define EXTERN_ASM 
@@ -374,8 +374,8 @@
 #define HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC 1
 #define HAVE_STRUCT_V4L2_FRMIVALENUM_DISCRETE 0
 #define HAVE_LIBDRM_GETFB2 0
-#define HAVE_MAKEINFO 1
-#define HAVE_MAKEINFO_HTML 1
+#define HAVE_MAKEINFO 0
+#define HAVE_MAKEINFO_HTML 0
 #define HAVE_OPENCL_D3D11 0
 #define HAVE_OPENCL_DRM_ARM 0
 #define HAVE_OPENCL_DRM_BEIGNET 0
@@ -425,7 +425,7 @@
 #define CONFIG_LIBXAVS2 0
 #define CONFIG_LIBXVID 0
 #define CONFIG_DECKLINK 0
-#define CONFIG_LIBFDK_AAC 0
+#define CONFIG_LIBFDK_AAC 1
 #define CONFIG_OPENSSL 0
 #define CONFIG_LIBTLS 0
 #define CONFIG_GMP 0
@@ -468,7 +468,7 @@
 #define CONFIG_LIBMP3LAME 0
 #define CONFIG_LIBMYSOFA 0
 #define CONFIG_LIBOPENCV 0
-#define CONFIG_LIBOPENH264 0
+#define CONFIG_LIBOPENH264 1
 #define CONFIG_LIBOPENJPEG 0
 #define CONFIG_LIBOPENMPT 0
 #define CONFIG_LIBOPENVINO 0
@@ -609,7 +609,7 @@
 #define CONFIG_PROTOCOLS 0
 #define CONFIG_AANDCTTABLES 0
 #define CONFIG_AC3DSP 0
-#define CONFIG_ADTS_HEADER 0
+#define CONFIG_ADTS_HEADER 1
 #define CONFIG_ATSC_A53 0
 #define CONFIG_AUDIO_FRAME_QUEUE 0
 #define CONFIG_AUDIODSP 0
@@ -709,7 +709,7 @@
 #define CONFIG_EXTRACT_EXTRADATA_BSF 0
 #define CONFIG_FILTER_UNITS_BSF 0
 #define CONFIG_H264_METADATA_BSF 0
-#define CONFIG_H264_MP4TOANNEXB_BSF 0
+#define CONFIG_H264_MP4TOANNEXB_BSF 1
 #define CONFIG_H264_REDUNDANT_PPS_BSF 0
 #define CONFIG_HAPQA_EXTRACT_BSF 0
 #define CONFIG_HEVC_METADATA_BSF 0
@@ -1241,7 +1241,7 @@
 #define CONFIG_LIBCODEC2_DECODER 0
 #define CONFIG_LIBDAV1D_DECODER 0
 #define CONFIG_LIBDAVS2_DECODER 0
-#define CONFIG_LIBFDK_AAC_DECODER 0
+#define CONFIG_LIBFDK_AAC_DECODER 1
 #define CONFIG_LIBGSM_DECODER 0
 #define CONFIG_LIBGSM_MS_DECODER 0
 #define CONFIG_LIBILBC_DECODER 0
@@ -1260,7 +1260,7 @@
 #define CONFIG_IDF_DECODER 0
 #define CONFIG_LIBAOM_AV1_DECODER 0
 #define CONFIG_AV1_DECODER 0
-#define CONFIG_LIBOPENH264_DECODER 0
+#define CONFIG_LIBOPENH264_DECODER 1
 #define CONFIG_H264_CUVID_DECODER 0
 #define CONFIG_HEVC_CUVID_DECODER 0
 #define CONFIG_HEVC_MEDIACODEC_DECODER 0
@@ -1561,7 +1561,7 @@
 #define CONFIG_WMV3_NVDEC_HWACCEL 0
 #define CONFIG_WMV3_VAAPI_HWACCEL 0
 #define CONFIG_WMV3_VDPAU_HWACCEL 0
-#define CONFIG_AAC_PARSER 0
+#define CONFIG_AAC_PARSER 1
 #define CONFIG_AAC_LATM_PARSER 0
 #define CONFIG_AC3_PARSER 0
 #define CONFIG_ADX_PARSER 0
@@ -2105,7 +2105,7 @@
 #define CONFIG_AFIFO_FILTER 0
 #define CONFIG_FIFO_FILTER 0
 #define CONFIG_AA_DEMUXER 0
-#define CONFIG_AAC_DEMUXER 0
+#define CONFIG_AAC_DEMUXER 1
 #define CONFIG_AAX_DEMUXER 0
 #define CONFIG_AC3_DEMUXER 0
 #define CONFIG_ACM_DEMUXER 0
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/bsf_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/bsf_list.c
index d31ece942a..d6f12718cf 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/bsf_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/bsf_list.c
@@ -1,3 +1,4 @@
 static const AVBitStreamFilter * const bitstream_filters[] = {
+    &ff_h264_mp4toannexb_bsf,
     &ff_null_bsf,
     NULL };
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/codec_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/codec_list.c
index 9407bd2775..ecd6c02721 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/codec_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/codec_list.c
@@ -14,5 +14,7 @@ static const AVCodec * const codec_list[] = {
     &ff_pcm_s24le_decoder,
     &ff_pcm_s32le_decoder,
     &ff_pcm_u8_decoder,
+    &ff_libfdk_aac_decoder,
     &ff_libopus_decoder,
+    &ff_libopenh264_decoder,
     NULL };
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/parser_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/parser_list.c
index f81fbe8bbc..7b86e2b934 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/parser_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavcodec/parser_list.c
@@ -1,4 +1,5 @@
 static const AVCodecParser * const parser_list[] = {
+    &ff_aac_parser,
     &ff_flac_parser,
     &ff_mpegaudio_parser,
     &ff_opus_parser,
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavformat/demuxer_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavformat/demuxer_list.c
index 1908ba19e7..920b22bfa7 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavformat/demuxer_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavformat/demuxer_list.c
@@ -1,4 +1,5 @@
 static const AVInputFormat * const demuxer_list[] = {
+    &ff_aac_demuxer,
     &ff_flac_demuxer,
     &ff_matroska_demuxer,
     &ff_mov_demuxer,
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavutil/ffversion.h b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavutil/ffversion.h
index 0f5c7b1d74..bc43b980d5 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavutil/ffversion.h
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/arm64/libavutil/ffversion.h
@@ -1,5 +1,5 @@
 /* Automatically generated by version.sh, do not manually edit! */
 #ifndef AVUTIL_FFVERSION_H
 #define AVUTIL_FFVERSION_H
-#define FFMPEG_VERSION "N-100481-g0493b979df"
+#define FFMPEG_VERSION "4.3.git"
 #endif /* AVUTIL_FFVERSION_H */
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.asm b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.asm
index f425351612..ae8255882f 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.asm
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.asm
@@ -54,7 +54,7 @@
 %define HAVE_SSE42 1
 %define HAVE_SSSE3 1
 %define HAVE_XOP 1
-%define HAVE_CPUNOP 0
+%define HAVE_CPUNOP 1
 %define HAVE_I686 1
 %define HAVE_MIPSFPU 0
 %define HAVE_MIPS32R2 0
@@ -88,7 +88,7 @@
 %define HAVE_AMD3DNOWEXT_EXTERNAL 1
 %define HAVE_AVX_EXTERNAL 1
 %define HAVE_AVX2_EXTERNAL 1
-%define HAVE_AVX512_EXTERNAL 1
+%define HAVE_AVX512_EXTERNAL 0
 %define HAVE_FMA3_EXTERNAL 1
 %define HAVE_FMA4_EXTERNAL 1
 %define HAVE_MMX_EXTERNAL 1
@@ -218,7 +218,7 @@
 %define HAVE_TERMIOS_H 1
 %define HAVE_UDPLITE_H 0
 %define HAVE_UNISTD_H 1
-%define HAVE_VALGRIND_VALGRIND_H 0 ; %define HAVE_VALGRIND_VALGRIND_H 1 -- forced to 0. See https://crbug.com/590440
+%define HAVE_VALGRIND_VALGRIND_H 0 ; %define HAVE_VALGRIND_VALGRIND_H 0 -- forced to 0. See https://crbug.com/590440
 %define HAVE_WINDOWS_H 0
 %define HAVE_WINSOCK2_H 0
 %define HAVE_INTRINSICS_NEON 0
@@ -359,8 +359,8 @@
 %define HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC 1
 %define HAVE_STRUCT_V4L2_FRMIVALENUM_DISCRETE 0
 %define HAVE_LIBDRM_GETFB2 0
-%define HAVE_MAKEINFO 1
-%define HAVE_MAKEINFO_HTML 1
+%define HAVE_MAKEINFO 0
+%define HAVE_MAKEINFO_HTML 0
 %define HAVE_OPENCL_D3D11 0
 %define HAVE_OPENCL_DRM_ARM 0
 %define HAVE_OPENCL_DRM_BEIGNET 0
@@ -410,7 +410,7 @@
 %define CONFIG_LIBXAVS2 0
 %define CONFIG_LIBXVID 0
 %define CONFIG_DECKLINK 0
-%define CONFIG_LIBFDK_AAC 0
+%define CONFIG_LIBFDK_AAC 1
 %define CONFIG_OPENSSL 0
 %define CONFIG_LIBTLS 0
 %define CONFIG_GMP 0
@@ -453,7 +453,7 @@
 %define CONFIG_LIBMP3LAME 0
 %define CONFIG_LIBMYSOFA 0
 %define CONFIG_LIBOPENCV 0
-%define CONFIG_LIBOPENH264 0
+%define CONFIG_LIBOPENH264 1
 %define CONFIG_LIBOPENJPEG 0
 %define CONFIG_LIBOPENMPT 0
 %define CONFIG_LIBOPENVINO 0
@@ -594,7 +594,7 @@
 %define CONFIG_PROTOCOLS 0
 %define CONFIG_AANDCTTABLES 0
 %define CONFIG_AC3DSP 0
-%define CONFIG_ADTS_HEADER 0
+%define CONFIG_ADTS_HEADER 1
 %define CONFIG_ATSC_A53 0
 %define CONFIG_AUDIO_FRAME_QUEUE 0
 %define CONFIG_AUDIODSP 0
@@ -694,7 +694,7 @@
 %define CONFIG_EXTRACT_EXTRADATA_BSF 0
 %define CONFIG_FILTER_UNITS_BSF 0
 %define CONFIG_H264_METADATA_BSF 0
-%define CONFIG_H264_MP4TOANNEXB_BSF 0
+%define CONFIG_H264_MP4TOANNEXB_BSF 1
 %define CONFIG_H264_REDUNDANT_PPS_BSF 0
 %define CONFIG_HAPQA_EXTRACT_BSF 0
 %define CONFIG_HEVC_METADATA_BSF 0
@@ -1226,7 +1226,7 @@
 %define CONFIG_LIBCODEC2_DECODER 0
 %define CONFIG_LIBDAV1D_DECODER 0
 %define CONFIG_LIBDAVS2_DECODER 0
-%define CONFIG_LIBFDK_AAC_DECODER 0
+%define CONFIG_LIBFDK_AAC_DECODER 1
 %define CONFIG_LIBGSM_DECODER 0
 %define CONFIG_LIBGSM_MS_DECODER 0
 %define CONFIG_LIBILBC_DECODER 0
@@ -1245,7 +1245,7 @@
 %define CONFIG_IDF_DECODER 0
 %define CONFIG_LIBAOM_AV1_DECODER 0
 %define CONFIG_AV1_DECODER 0
-%define CONFIG_LIBOPENH264_DECODER 0
+%define CONFIG_LIBOPENH264_DECODER 1
 %define CONFIG_H264_CUVID_DECODER 0
 %define CONFIG_HEVC_CUVID_DECODER 0
 %define CONFIG_HEVC_MEDIACODEC_DECODER 0
@@ -1546,7 +1546,7 @@
 %define CONFIG_WMV3_NVDEC_HWACCEL 0
 %define CONFIG_WMV3_VAAPI_HWACCEL 0
 %define CONFIG_WMV3_VDPAU_HWACCEL 0
-%define CONFIG_AAC_PARSER 0
+%define CONFIG_AAC_PARSER 1
 %define CONFIG_AAC_LATM_PARSER 0
 %define CONFIG_AC3_PARSER 0
 %define CONFIG_ADX_PARSER 0
@@ -2090,7 +2090,7 @@
 %define CONFIG_AFIFO_FILTER 0
 %define CONFIG_FIFO_FILTER 0
 %define CONFIG_AA_DEMUXER 0
-%define CONFIG_AAC_DEMUXER 0
+%define CONFIG_AAC_DEMUXER 1
 %define CONFIG_AAX_DEMUXER 0
 %define CONFIG_AC3_DEMUXER 0
 %define CONFIG_ACM_DEMUXER 0
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.h b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.h
index b04ca4da02..81eee19918 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.h
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/config.h
@@ -1,12 +1,12 @@
 /* Automatically generated by configure - do not modify! */
 #ifndef FFMPEG_CONFIG_H
 #define FFMPEG_CONFIG_H
-/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-fft --enable-rdft --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-lzo --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/mnt/src/chromium/src/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --enable-decoder='theora,vp8' --enable-parser='vp3,vp8' --enable-lto --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld'" -- elide long configuration string from binary */
+/* #define FFMPEG_CONFIGURATION "--disable-everything --disable-all --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-static --enable-avcodec --enable-avformat --enable-avutil --enable-fft --enable-rdft --enable-static --enable-libopus --disable-debug --disable-bzlib --disable-error-resilience --disable-iconv --disable-lzo --disable-network --disable-schannel --disable-sdl2 --disable-symver --disable-xlib --disable-zlib --disable-securetransport --disable-faan --disable-alsa --disable-autodetect --enable-decoder='vorbis,libopus,flac' --enable-decoder='pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,mp3' --enable-decoder='pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw' --enable-demuxer='ogg,matroska,wav,flac,mp3,mov' --enable-parser='opus,vorbis,flac,mpegaudio,vp9' --extra-cflags=-I/chromium/third_party/opus/src/include --disable-linux-perf --x86asmexe=nasm --optflags='\"-O2\"' --enable-decoder='theora,vp8' --enable-parser='vp3,vp8' --enable-lto --enable-pic --cc=clang --cxx=clang++ --ld=clang --extra-ldflags='-fuse-ld=lld' --enable-libfdk-aac --enable-decoder=libfdk_aac --enable-demuxer=aac --enable-parser=aac --enable-libopenh264 --enable-decoder=libopenh264" -- elide long configuration string from binary */
 #define FFMPEG_LICENSE "LGPL version 2.1 or later"
 #define CONFIG_THIS_YEAR 2020
 #define FFMPEG_DATADIR "/usr/local/share/ffmpeg"
 #define AVCONV_DATADIR "/usr/local/share/ffmpeg"
-#define CC_IDENT "clang version 12.0.0 (https://github.com/llvm/llvm-project/ d0abc757495349fd053beeaea81cd954c2e457e7)"
+#define CC_IDENT "clang version 12.0.0 (https://github.com/llvm/llvm-project/ 418f18c6cdfe56e77669e2f4d3df3bca1020156d)"
 #define av_restrict restrict
 #define EXTERN_PREFIX ""
 #define EXTERN_ASM 
@@ -69,7 +69,7 @@
 #define HAVE_SSE42 1
 #define HAVE_SSSE3 1
 #define HAVE_XOP 1
-#define HAVE_CPUNOP 0
+#define HAVE_CPUNOP 1
 #define HAVE_I686 1
 #define HAVE_MIPSFPU 0
 #define HAVE_MIPS32R2 0
@@ -103,7 +103,7 @@
 #define HAVE_AMD3DNOWEXT_EXTERNAL 1
 #define HAVE_AVX_EXTERNAL 1
 #define HAVE_AVX2_EXTERNAL 1
-#define HAVE_AVX512_EXTERNAL 1
+#define HAVE_AVX512_EXTERNAL 0
 #define HAVE_FMA3_EXTERNAL 1
 #define HAVE_FMA4_EXTERNAL 1
 #define HAVE_MMX_EXTERNAL 1
@@ -233,7 +233,7 @@
 #define HAVE_TERMIOS_H 1
 #define HAVE_UDPLITE_H 0
 #define HAVE_UNISTD_H 1
-#define HAVE_VALGRIND_VALGRIND_H 0 /* #define HAVE_VALGRIND_VALGRIND_H 1 -- forced to 0. See https://crbug.com/590440 */
+#define HAVE_VALGRIND_VALGRIND_H 0 /* #define HAVE_VALGRIND_VALGRIND_H 0 -- forced to 0. See https://crbug.com/590440 */
 #define HAVE_WINDOWS_H 0
 #define HAVE_WINSOCK2_H 0
 #define HAVE_INTRINSICS_NEON 0
@@ -374,8 +374,8 @@
 #define HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC 1
 #define HAVE_STRUCT_V4L2_FRMIVALENUM_DISCRETE 0
 #define HAVE_LIBDRM_GETFB2 0
-#define HAVE_MAKEINFO 1
-#define HAVE_MAKEINFO_HTML 1
+#define HAVE_MAKEINFO 0
+#define HAVE_MAKEINFO_HTML 0
 #define HAVE_OPENCL_D3D11 0
 #define HAVE_OPENCL_DRM_ARM 0
 #define HAVE_OPENCL_DRM_BEIGNET 0
@@ -425,7 +425,7 @@
 #define CONFIG_LIBXAVS2 0
 #define CONFIG_LIBXVID 0
 #define CONFIG_DECKLINK 0
-#define CONFIG_LIBFDK_AAC 0
+#define CONFIG_LIBFDK_AAC 1
 #define CONFIG_OPENSSL 0
 #define CONFIG_LIBTLS 0
 #define CONFIG_GMP 0
@@ -468,7 +468,7 @@
 #define CONFIG_LIBMP3LAME 0
 #define CONFIG_LIBMYSOFA 0
 #define CONFIG_LIBOPENCV 0
-#define CONFIG_LIBOPENH264 0
+#define CONFIG_LIBOPENH264 1
 #define CONFIG_LIBOPENJPEG 0
 #define CONFIG_LIBOPENMPT 0
 #define CONFIG_LIBOPENVINO 0
@@ -609,7 +609,7 @@
 #define CONFIG_PROTOCOLS 0
 #define CONFIG_AANDCTTABLES 0
 #define CONFIG_AC3DSP 0
-#define CONFIG_ADTS_HEADER 0
+#define CONFIG_ADTS_HEADER 1
 #define CONFIG_ATSC_A53 0
 #define CONFIG_AUDIO_FRAME_QUEUE 0
 #define CONFIG_AUDIODSP 0
@@ -709,7 +709,7 @@
 #define CONFIG_EXTRACT_EXTRADATA_BSF 0
 #define CONFIG_FILTER_UNITS_BSF 0
 #define CONFIG_H264_METADATA_BSF 0
-#define CONFIG_H264_MP4TOANNEXB_BSF 0
+#define CONFIG_H264_MP4TOANNEXB_BSF 1
 #define CONFIG_H264_REDUNDANT_PPS_BSF 0
 #define CONFIG_HAPQA_EXTRACT_BSF 0
 #define CONFIG_HEVC_METADATA_BSF 0
@@ -1241,7 +1241,7 @@
 #define CONFIG_LIBCODEC2_DECODER 0
 #define CONFIG_LIBDAV1D_DECODER 0
 #define CONFIG_LIBDAVS2_DECODER 0
-#define CONFIG_LIBFDK_AAC_DECODER 0
+#define CONFIG_LIBFDK_AAC_DECODER 1
 #define CONFIG_LIBGSM_DECODER 0
 #define CONFIG_LIBGSM_MS_DECODER 0
 #define CONFIG_LIBILBC_DECODER 0
@@ -1260,7 +1260,7 @@
 #define CONFIG_IDF_DECODER 0
 #define CONFIG_LIBAOM_AV1_DECODER 0
 #define CONFIG_AV1_DECODER 0
-#define CONFIG_LIBOPENH264_DECODER 0
+#define CONFIG_LIBOPENH264_DECODER 1
 #define CONFIG_H264_CUVID_DECODER 0
 #define CONFIG_HEVC_CUVID_DECODER 0
 #define CONFIG_HEVC_MEDIACODEC_DECODER 0
@@ -1561,7 +1561,7 @@
 #define CONFIG_WMV3_NVDEC_HWACCEL 0
 #define CONFIG_WMV3_VAAPI_HWACCEL 0
 #define CONFIG_WMV3_VDPAU_HWACCEL 0
-#define CONFIG_AAC_PARSER 0
+#define CONFIG_AAC_PARSER 1
 #define CONFIG_AAC_LATM_PARSER 0
 #define CONFIG_AC3_PARSER 0
 #define CONFIG_ADX_PARSER 0
@@ -2105,7 +2105,7 @@
 #define CONFIG_AFIFO_FILTER 0
 #define CONFIG_FIFO_FILTER 0
 #define CONFIG_AA_DEMUXER 0
-#define CONFIG_AAC_DEMUXER 0
+#define CONFIG_AAC_DEMUXER 1
 #define CONFIG_AAX_DEMUXER 0
 #define CONFIG_AC3_DEMUXER 0
 #define CONFIG_ACM_DEMUXER 0
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/bsf_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/bsf_list.c
index d31ece942a..d6f12718cf 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/bsf_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/bsf_list.c
@@ -1,3 +1,4 @@
 static const AVBitStreamFilter * const bitstream_filters[] = {
+    &ff_h264_mp4toannexb_bsf,
     &ff_null_bsf,
     NULL };
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/codec_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/codec_list.c
index 9407bd2775..ecd6c02721 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/codec_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/codec_list.c
@@ -14,5 +14,7 @@ static const AVCodec * const codec_list[] = {
     &ff_pcm_s24le_decoder,
     &ff_pcm_s32le_decoder,
     &ff_pcm_u8_decoder,
+    &ff_libfdk_aac_decoder,
     &ff_libopus_decoder,
+    &ff_libopenh264_decoder,
     NULL };
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/parser_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/parser_list.c
index f81fbe8bbc..7b86e2b934 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/parser_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavcodec/parser_list.c
@@ -1,4 +1,5 @@
 static const AVCodecParser * const parser_list[] = {
+    &ff_aac_parser,
     &ff_flac_parser,
     &ff_mpegaudio_parser,
     &ff_opus_parser,
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavformat/demuxer_list.c b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavformat/demuxer_list.c
index 1908ba19e7..920b22bfa7 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavformat/demuxer_list.c
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavformat/demuxer_list.c
@@ -1,4 +1,5 @@
 static const AVInputFormat * const demuxer_list[] = {
+    &ff_aac_demuxer,
     &ff_flac_demuxer,
     &ff_matroska_demuxer,
     &ff_mov_demuxer,
diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
index 0f5c7b1d74..bc43b980d5 100644
--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
+++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
@@ -1,5 +1,5 @@
 /* Automatically generated by version.sh, do not manually edit! */
 #ifndef AVUTIL_FFVERSION_H
 #define AVUTIL_FFVERSION_H
-#define FFMPEG_VERSION "N-100481-g0493b979df"
+#define FFMPEG_VERSION "4.3.git"
 #endif /* AVUTIL_FFVERSION_H */
diff --git a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
index 75333a2c9f..3b59e28c25 100755
--- a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
+++ b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
@@ -910,11 +910,22 @@ def ConfigureAndBuild(target_arch, target_os, host_os, host_arch, parallel_jobs,
           '--ar=cygwin-wrapper lib',
       ])
 
+  # Depend on royalty-free libfdk-aac for AAC support.
+  if target_os == 'linux':
+    configure_flags['Common'].extend([
+        # Build against libfdk-aac and enable it always.
+        '--enable-libfdk-aac',
+        '--enable-decoder=libfdk_aac',
+        '--enable-demuxer=aac',
+        '--enable-parser=aac',
+        '--enable-libopenh264',
+        '--enable-decoder=libopenh264',
+    ])
+
   # Google Chrome & ChromeOS specific configuration.
   configure_flags['Chrome'].extend([
-      '--enable-decoder=aac,h264',
-      '--enable-demuxer=aac',
-      '--enable-parser=aac,h264',
+      '--enable-decoder=h264',
+      '--enable-parser=h264',
   ])
 
   # Google ChromeOS specific configuration.
diff --git a/third_party/ffmpeg/ffmpeg_generated.gni b/third_party/ffmpeg/ffmpeg_generated.gni
index 71f17d1735..0e6f1f516a 100644
--- a/third_party/ffmpeg/ffmpeg_generated.gni
+++ b/third_party/ffmpeg/ffmpeg_generated.gni
@@ -1,4 +1,4 @@
-# Copyright 2020 The Chromium Authors. All rights reserved.
+# Copyright 2021 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -14,17 +14,24 @@ ffmpeg_asm_sources = []
 
 use_linux_config = is_linux || is_chromeos || is_fuchsia
 
-if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && current_cpu == "arm64") || (is_android && current_cpu == "x64") || (is_android && current_cpu == "x86") || (is_mac) || (is_win) || (use_linux_config)) {
+if ((use_linux_config && current_cpu == "arm64") || (use_linux_config && current_cpu == "x64")) {
   ffmpeg_c_sources += [
+    "libavcodec/aac_ac3_parser.c",
+    "libavcodec/aac_parser.c",
     "libavcodec/ac3_parser.c",
     "libavcodec/ac3tab.c",
+    "libavcodec/adts_header.c",
     "libavcodec/adts_parser.c",
     "libavcodec/allcodecs.c",
     "libavcodec/autorename_libavcodec_flacdec.c",
     "libavcodec/autorename_libavcodec_flacdsp.c",
+    "libavcodec/autorename_libavcodec_hpeldsp.c",
     "libavcodec/autorename_libavcodec_mpegaudiodsp.c",
     "libavcodec/autorename_libavcodec_utils.c",
+    "libavcodec/autorename_libavcodec_videodsp.c",
     "libavcodec/autorename_libavcodec_vorbisdsp.c",
+    "libavcodec/autorename_libavcodec_vp3dsp.c",
+    "libavcodec/autorename_libavcodec_vp8dsp.c",
     "libavcodec/avdct.c",
     "libavcodec/avfft.c",
     "libavcodec/avpacket.c",
@@ -50,8 +57,13 @@ if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && curre
     "libavcodec/flac_parser.c",
     "libavcodec/flacdata.c",
     "libavcodec/golomb.c",
+    "libavcodec/h264_mp4toannexb_bsf.c",
+    "libavcodec/h264pred.c",
     "libavcodec/imgconvert.c",
     "libavcodec/jni.c",
+    "libavcodec/libfdk-aacdec.c",
+    "libavcodec/libopenh264.c",
+    "libavcodec/libopenh264dec.c",
     "libavcodec/libopus.c",
     "libavcodec/libopusdec.c",
     "libavcodec/mathtables.c",
@@ -90,9 +102,16 @@ if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && curre
     "libavcodec/vorbis_data.c",
     "libavcodec/vorbis_parser.c",
     "libavcodec/vorbisdec.c",
+    "libavcodec/vp3.c",
+    "libavcodec/vp3_parser.c",
+    "libavcodec/vp56rac.c",
+    "libavcodec/vp8.c",
+    "libavcodec/vp8_parser.c",
     "libavcodec/vp9_parser.c",
     "libavcodec/xiph.c",
+    "libavformat/aacdec.c",
     "libavformat/allformats.c",
+    "libavformat/apetag.c",
     "libavformat/autorename_libavformat_options.c",
     "libavformat/autorename_libavformat_pcm.c",
     "libavformat/autorename_libavformat_utils.c",
@@ -105,6 +124,7 @@ if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && curre
     "libavformat/format.c",
     "libavformat/id3v1.c",
     "libavformat/id3v2.c",
+    "libavformat/img2.c",
     "libavformat/isom.c",
     "libavformat/matroska.c",
     "libavformat/matroskadec.c",
@@ -195,44 +215,7 @@ if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && curre
   ]
 }
 
-if ((is_mac) || (is_win) || (use_linux_config)) {
-  ffmpeg_c_sources += [
-    "libavcodec/autorename_libavcodec_hpeldsp.c",
-    "libavcodec/autorename_libavcodec_videodsp.c",
-    "libavcodec/autorename_libavcodec_vp3dsp.c",
-    "libavcodec/autorename_libavcodec_vp8dsp.c",
-    "libavcodec/h264pred.c",
-    "libavcodec/vp3.c",
-    "libavcodec/vp3_parser.c",
-    "libavcodec/vp56rac.c",
-    "libavcodec/vp8.c",
-    "libavcodec/vp8_parser.c",
-  ]
-}
-
-if ((current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_android && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (is_android && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (is_mac && ffmpeg_branding == "Chrome") || (is_win && ffmpeg_branding == "Chrome") || (use_linux_config && ffmpeg_branding == "Chrome") || (use_linux_config && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/aac_ac3_parser.c",
-    "libavcodec/aac_parser.c",
-    "libavcodec/aacps_float.c",
-    "libavcodec/aacpsdsp_float.c",
-    "libavcodec/aacsbr.c",
-    "libavcodec/aactab.c",
-    "libavcodec/adts_header.c",
-    "libavcodec/autorename_libavcodec_aacdec.c",
-    "libavcodec/autorename_libavcodec_mdct15.c",
-    "libavcodec/autorename_libavcodec_sbrdsp.c",
-    "libavcodec/cbrt_data.c",
-    "libavcodec/kbdwin.c",
-    "libavcodec/sinewin.c",
-    "libavcodec/sinewin_fixed.c",
-    "libavformat/aacdec.c",
-    "libavformat/apetag.c",
-    "libavformat/img2.c",
-  ]
-}
-
-if ((is_mac && ffmpeg_branding == "Chrome") || (is_win && ffmpeg_branding == "Chrome") || (use_linux_config && ffmpeg_branding == "Chrome") || (use_linux_config && ffmpeg_branding == "ChromeOS")) {
+if ((use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_c_sources += [
     "libavcodec/atsc_a53.c",
     "libavcodec/cabac.c",
@@ -259,120 +242,89 @@ if ((is_mac && ffmpeg_branding == "Chrome") || (is_win && ffmpeg_branding == "Ch
   ]
 }
 
-if ((is_android && current_cpu == "x64") || (is_android && current_cpu == "x86") || (is_mac && current_cpu == "x64") || (is_win && current_cpu == "x64") || (is_win && current_cpu == "x86") || (use_linux_config && current_cpu == "x64") || (use_linux_config && current_cpu == "x86")) {
-  ffmpeg_c_sources += [
-    "libavcodec/x86/autorename_libavcodec_x86_vorbisdsp_init.c",
-    "libavcodec/x86/constants.c",
-    "libavcodec/x86/dct_init.c",
-    "libavcodec/x86/fft_init.c",
-    "libavcodec/x86/flacdsp_init.c",
-    "libavcodec/x86/mpegaudiodsp.c",
-    "libavutil/x86/autorename_libavutil_x86_cpu.c",
-    "libavutil/x86/autorename_libavutil_x86_float_dsp_init.c",
-    "libavutil/x86/fixed_dsp_init.c",
-    "libavutil/x86/imgutils_init.c",
-    "libavutil/x86/lls_init.c",
-  ]
-}
-
-if ((is_android && current_cpu == "arm64") || (is_mac && current_cpu == "arm64") || (is_win && current_cpu == "arm64") || (use_linux_config && current_cpu == "arm64")) {
+if (use_linux_config && current_cpu == "arm64") {
   ffmpeg_c_sources += [
+    "libavcodec/aarch64/autorename_libavcodec_aarch64_videodsp_init.c",
+    "libavcodec/aarch64/autorename_libavcodec_aarch64_vorbisdsp_init.c",
     "libavcodec/aarch64/fft_init_aarch64.c",
+    "libavcodec/aarch64/h264pred_init.c",
+    "libavcodec/aarch64/hpeldsp_init_aarch64.c",
     "libavcodec/aarch64/mpegaudiodsp_init.c",
-    "libavcodec/aarch64/vorbisdsp_init.c",
+    "libavcodec/aarch64/vp8dsp_init_aarch64.c",
     "libavutil/aarch64/autorename_libavutil_aarch64_cpu.c",
-    "libavutil/aarch64/float_dsp_init.c",
+    "libavutil/aarch64/autorename_libavutil_aarch64_float_dsp_init.c",
   ]
   ffmpeg_gas_sources += [
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_fft_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_mdct_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_vorbisdsp_neon.S",
+    "libavcodec/aarch64/autorename_libavcodec_aarch64_videodsp.S",
+    "libavcodec/aarch64/fft_neon.S",
+    "libavcodec/aarch64/h264pred_neon.S",
+    "libavcodec/aarch64/hpeldsp_neon.S",
+    "libavcodec/aarch64/mdct_neon.S",
     "libavcodec/aarch64/mpegaudiodsp_neon.S",
-    "libavutil/aarch64/autorename_libavutil_aarch64_float_dsp_neon.S",
+    "libavcodec/aarch64/vorbisdsp_neon.S",
+    "libavcodec/aarch64/vp8dsp_neon.S",
+    "libavutil/aarch64/float_dsp_neon.S",
   ]
 }
 
-if ((is_android && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm")) {
+if ((use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_c_sources += [
-    "libavcodec/arm/fft_fixed_init_arm.c",
-    "libavcodec/arm/fft_init_arm.c",
-    "libavcodec/arm/flacdsp_init_arm.c",
-    "libavcodec/arm/mpegaudiodsp_init_arm.c",
-    "libavcodec/arm/rdft_init_arm.c",
-    "libavcodec/arm/vorbisdsp_init_arm.c",
-    "libavutil/arm/autorename_libavutil_arm_cpu.c",
-    "libavutil/arm/float_dsp_init_arm.c",
-    "libavutil/arm/float_dsp_init_vfp.c",
+    "libavcodec/aarch64/h264chroma_init_aarch64.c",
+    "libavcodec/aarch64/h264dsp_init_aarch64.c",
+    "libavcodec/aarch64/h264qpel_init_aarch64.c",
   ]
   ffmpeg_gas_sources += [
-    "libavcodec/arm/fft_vfp.S",
-    "libavcodec/arm/flacdsp_arm.S",
-    "libavcodec/arm/mdct_vfp.S",
-    "libavcodec/arm/mpegaudiodsp_fixed_armv6.S",
-    "libavutil/arm/float_dsp_vfp.S",
-  ]
-}
-
-if ((is_android && current_cpu == "x64") || (is_mac && current_cpu == "x64") || (is_win && current_cpu == "x64") || (is_win && current_cpu == "x86") || (use_linux_config && current_cpu == "x64") || (use_linux_config && current_cpu == "x86")) {
-  ffmpeg_asm_sources += [
-    "libavcodec/x86/dct32.asm",
-    "libavcodec/x86/fft.asm",
-    "libavcodec/x86/flacdsp.asm",
-    "libavcodec/x86/imdct36.asm",
-    "libavcodec/x86/vorbisdsp.asm",
-    "libavutil/x86/cpuid.asm",
-    "libavutil/x86/fixed_dsp.asm",
-    "libavutil/x86/float_dsp.asm",
-    "libavutil/x86/imgutils.asm",
-    "libavutil/x86/lls.asm",
+    "libavcodec/aarch64/h264cmc_neon.S",
+    "libavcodec/aarch64/h264dsp_neon.S",
+    "libavcodec/aarch64/h264idct_neon.S",
+    "libavcodec/aarch64/h264qpel_neon.S",
   ]
 }
 
-if ((is_mac && current_cpu == "x64") || (is_win && current_cpu == "x64") || (is_win && current_cpu == "x86") || (use_linux_config && current_cpu == "x64") || (use_linux_config && current_cpu == "x86")) {
+if (use_linux_config && current_cpu == "x64") {
   ffmpeg_c_sources += [
-    "libavcodec/x86/autorename_libavcodec_x86_videodsp_init.c",
+    "libavcodec/x86/constants.c",
+    "libavcodec/x86/dct_init.c",
+    "libavcodec/x86/fft_init.c",
+    "libavcodec/x86/flacdsp_init.c",
     "libavcodec/x86/h264_intrapred_init.c",
     "libavcodec/x86/hpeldsp_init.c",
     "libavcodec/x86/hpeldsp_vp3_init.c",
+    "libavcodec/x86/mpegaudiodsp.c",
+    "libavcodec/x86/videodsp_init.c",
+    "libavcodec/x86/vorbisdsp_init.c",
     "libavcodec/x86/vp3dsp_init.c",
     "libavcodec/x86/vp8dsp_init.c",
+    "libavutil/x86/cpu.c",
+    "libavutil/x86/fixed_dsp_init.c",
+    "libavutil/x86/float_dsp_init.c",
+    "libavutil/x86/imgutils_init.c",
+    "libavutil/x86/lls_init.c",
   ]
   ffmpeg_asm_sources += [
-    "libavcodec/x86/autorename_libavcodec_x86_videodsp.asm",
+    "libavcodec/x86/dct32.asm",
+    "libavcodec/x86/fft.asm",
+    "libavcodec/x86/flacdsp.asm",
     "libavcodec/x86/fpel.asm",
     "libavcodec/x86/h264_intrapred.asm",
     "libavcodec/x86/h264_intrapred_10bit.asm",
     "libavcodec/x86/hpeldsp.asm",
     "libavcodec/x86/hpeldsp_vp3.asm",
+    "libavcodec/x86/imdct36.asm",
+    "libavcodec/x86/videodsp.asm",
+    "libavcodec/x86/vorbisdsp.asm",
     "libavcodec/x86/vp3dsp.asm",
     "libavcodec/x86/vp8dsp.asm",
     "libavcodec/x86/vp8dsp_loopfilter.asm",
+    "libavutil/x86/cpuid.asm",
+    "libavutil/x86/fixed_dsp.asm",
+    "libavutil/x86/float_dsp.asm",
+    "libavutil/x86/imgutils.asm",
+    "libavutil/x86/lls.asm",
   ]
 }
 
-if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_android && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/x86/aacpsdsp_init.c",
-    "libavcodec/x86/mdct15_init.c",
-    "libavcodec/x86/sbrdsp_init.c",
-  ]
-}
-
-if ((is_android && current_cpu == "arm" && arm_use_neon) || (is_android && current_cpu == "arm64") || (is_android && current_cpu == "x64") || (is_android && current_cpu == "x86")) {
-  ffmpeg_c_sources += [
-    "compat/strtod.c",
-  ]
-}
-
-if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_asm_sources += [
-    "libavcodec/x86/aacpsdsp.asm",
-    "libavcodec/x86/mdct15.asm",
-    "libavcodec/x86/sbrdsp.asm",
-  ]
-}
-
-if (use_linux_config && ffmpeg_branding == "ChromeOS") {
+if ((use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_c_sources += [
     "libavcodec/acelp_filters.c",
     "libavcodec/acelp_pitch_delay.c",
@@ -424,112 +376,7 @@ if (use_linux_config && ffmpeg_branding == "ChromeOS") {
   ]
 }
 
-if ((use_linux_config && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm")) {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/h264pred_init_arm.c",
-    "libavcodec/arm/hpeldsp_init_arm.c",
-    "libavcodec/arm/hpeldsp_init_armv6.c",
-    "libavcodec/arm/videodsp_init_arm.c",
-    "libavcodec/arm/videodsp_init_armv5te.c",
-    "libavcodec/arm/vp3dsp_init_arm.c",
-    "libavcodec/arm/vp8dsp_init_arm.c",
-    "libavcodec/arm/vp8dsp_init_armv6.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/hpeldsp_arm.S",
-    "libavcodec/arm/hpeldsp_armv6.S",
-    "libavcodec/arm/videodsp_armv5te.S",
-    "libavcodec/arm/vp8_armv6.S",
-    "libavcodec/arm/vp8dsp_armv6.S",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "mips64el") || (use_linux_config && current_cpu == "mipsel")) {
-  ffmpeg_c_sources += [
-    "libavcodec/mips/autorename_libavcodec_mips_videodsp_init.c",
-    "libavcodec/mips/h264pred_init_mips.c",
-    "libavcodec/mips/hpeldsp_init_mips.c",
-    "libavcodec/mips/vp3dsp_init_mips.c",
-    "libavcodec/mips/vp8dsp_init_mips.c",
-    "libavutil/mips/cpu.c",
-    "libavutil/mips/float_dsp_mips.c",
-  ]
-}
-
-if ((is_mac && current_cpu == "arm64") || (is_win && current_cpu == "arm64") || (use_linux_config && current_cpu == "arm64")) {
-  ffmpeg_c_sources += [
-    "libavcodec/aarch64/h264pred_init.c",
-    "libavcodec/aarch64/hpeldsp_init_aarch64.c",
-    "libavcodec/aarch64/videodsp_init.c",
-    "libavcodec/aarch64/vp8dsp_init_aarch64.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_h264pred_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_hpeldsp_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_vp8dsp_neon.S",
-    "libavcodec/aarch64/videodsp.S",
-  ]
-}
-
-if ((current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/aarch64/aacpsdsp_init_aarch64.c",
-    "libavcodec/aarch64/sbrdsp_init_aarch64.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_aacpsdsp_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_sbrdsp_neon.S",
-  ]
-}
-
-if ((is_mac && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/aarch64/h264chroma_init_aarch64.c",
-    "libavcodec/aarch64/h264dsp_init_aarch64.c",
-    "libavcodec/aarch64/h264qpel_init_aarch64.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_h264cmc_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_h264dsp_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_h264idct_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_h264qpel_neon.S",
-  ]
-}
-
-if ((is_android && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "arm" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/aacpsdsp_init_arm.c",
-    "libavcodec/arm/sbrdsp_init_arm.c",
-  ]
-}
-
-if ((is_android && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm" && arm_use_neon)) {
-  ffmpeg_c_sources += [
-    "libavutil/arm/float_dsp_init_neon.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/fft_fixed_neon.S",
-    "libavcodec/arm/fft_neon.S",
-    "libavcodec/arm/mdct_fixed_neon.S",
-    "libavcodec/arm/mdct_neon.S",
-    "libavcodec/arm/rdft_neon.S",
-    "libavcodec/arm/vorbisdsp_neon.S",
-    "libavutil/arm/float_dsp_neon.S",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "arm" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/h264chroma_init_arm.c",
-    "libavcodec/arm/h264dsp_init_arm.c",
-    "libavcodec/arm/h264qpel_init_arm.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/startcode_armv6.S",
-  ]
-}
-
-if ((is_mac && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "ChromeOS")) {
+if ((use_linux_config && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_c_sources += [
     "libavcodec/x86/h264_qpel.c",
     "libavcodec/x86/h264chroma_init.c",
@@ -550,98 +397,19 @@ if ((is_mac && current_cpu == "x64" && ffmpeg_branding == "Chrome") || (is_win &
   ]
 }
 
-if ((use_linux_config && current_cpu == "mips64el" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "mips64el" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "mipsel" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "mipsel" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/mips/aacdec_mips.c",
-    "libavcodec/mips/aacpsdsp_mips.c",
-    "libavcodec/mips/aacsbr_mips.c",
-    "libavcodec/mips/h264chroma_init_mips.c",
-    "libavcodec/mips/h264dsp_init_mips.c",
-    "libavcodec/mips/h264qpel_init_mips.c",
-    "libavcodec/mips/sbrdsp_mips.c",
-  ]
-}
-
-if (use_linux_config && current_cpu == "arm" && arm_use_neon) {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/hpeldsp_init_neon.c",
-    "libavcodec/arm/vp8dsp_init_neon.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/h264pred_neon.S",
-    "libavcodec/arm/hpeldsp_neon.S",
-    "libavcodec/arm/vp3dsp_neon.S",
-    "libavcodec/arm/vp8dsp_neon.S",
-  ]
-}
-
-if ((is_android && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/aacpsdsp_neon.S",
-    "libavcodec/arm/sbrdsp_neon.S",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "arm" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/blockdsp_init_arm.c",
-    "libavcodec/arm/idctdsp_init_arm.c",
-    "libavcodec/arm/idctdsp_init_armv5te.c",
-    "libavcodec/arm/idctdsp_init_armv6.c",
-    "libavcodec/arm/me_cmp_init_arm.c",
-    "libavcodec/arm/mpegvideo_arm.c",
-    "libavcodec/arm/mpegvideo_armv5te.c",
-    "libavcodec/arm/pixblockdsp_init_arm.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/idctdsp_arm.S",
-    "libavcodec/arm/idctdsp_armv6.S",
-    "libavcodec/arm/jrevdct_arm.S",
-    "libavcodec/arm/me_cmp_armv6.S",
-    "libavcodec/arm/mpegvideo_armv5te_s.S",
-    "libavcodec/arm/pixblockdsp_armv6.S",
-    "libavcodec/arm/simple_idct_arm.S",
-    "libavcodec/arm/simple_idct_armv5te.S",
-    "libavcodec/arm/simple_idct_armv6.S",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS")) {
+if (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS") {
   ffmpeg_c_sources += [
+    "libavcodec/aarch64/idctdsp_init_aarch64.c",
+    "libavcodec/aarch64/pixblockdsp_init_aarch64.c",
     "libavcodec/neon/autorename_libavcodec_neon_mpegvideo.c",
   ]
-}
-
-if ((use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_gas_sources += [
-    "libavcodec/arm/h264cmc_neon.S",
-    "libavcodec/arm/h264dsp_neon.S",
-    "libavcodec/arm/h264idct_neon.S",
-    "libavcodec/arm/h264qpel_neon.S",
+    "libavcodec/aarch64/pixblockdsp_neon.S",
+    "libavcodec/aarch64/simple_idct_neon.S",
   ]
 }
 
-if (use_linux_config && current_cpu == "mips64el") {
-  ffmpeg_c_sources += [
-    "libavcodec/mips/fft_mips.c",
-    "libavcodec/mips/mpegaudiodsp_mips_float.c",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "mips64el" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "mipsel" && ffmpeg_branding == "ChromeOS")) {
-  ffmpeg_c_sources += [
-    "libavcodec/mips/blockdsp_init_mips.c",
-    "libavcodec/mips/h263dsp_init_mips.c",
-    "libavcodec/mips/idctdsp_init_mips.c",
-    "libavcodec/mips/me_cmp_init_mips.c",
-    "libavcodec/mips/mpegvideo_init_mips.c",
-    "libavcodec/mips/pixblockdsp_init_mips.c",
-    "libavcodec/mips/qpeldsp_init_mips.c",
-    "libavcodec/mips/xvididct_init_mips.c",
-  ]
-}
-
-if ((use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS") || (use_linux_config && current_cpu == "x86" && ffmpeg_branding == "ChromeOS")) {
+if (use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS") {
   ffmpeg_c_sources += [
     "libavcodec/x86/blockdsp_init.c",
     "libavcodec/x86/fdct.c",
@@ -668,38 +436,3 @@ if ((use_linux_config && current_cpu == "x64" && ffmpeg_branding == "ChromeOS")
   ]
 }
 
-if (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS") {
-  ffmpeg_c_sources += [
-    "libavcodec/aarch64/idctdsp_init_aarch64.c",
-    "libavcodec/aarch64/pixblockdsp_init_aarch64.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_pixblockdsp_neon.S",
-    "libavcodec/aarch64/autorename_libavcodec_aarch64_simple_idct_neon.S",
-  ]
-}
-
-if (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") {
-  ffmpeg_c_sources += [
-    "libavcodec/arm/blockdsp_init_neon.c",
-    "libavcodec/arm/idctdsp_init_neon.c",
-  ]
-  ffmpeg_gas_sources += [
-    "libavcodec/arm/blockdsp_neon.S",
-    "libavcodec/arm/idctdsp_neon.S",
-    "libavcodec/arm/mpegvideo_neon.S",
-    "libavcodec/arm/pixblockdsp_neon.S",
-    "libavcodec/arm/simple_idct_neon.S",
-  ]
-}
-
-if (use_linux_config && current_cpu == "mips64el" && ffmpeg_branding == "ChromeOS") {
-  ffmpeg_c_sources += [
-    "libavcodec/mips/acelp_filters_mips.c",
-    "libavcodec/mips/acelp_vectors_mips.c",
-    "libavcodec/mips/amrwbdec_mips.c",
-    "libavcodec/mips/celp_filters_mips.c",
-    "libavcodec/mips/celp_math_mips.c",
-  ]
-}
-
-- 
2.29.2

